Source/objects.cpp

```diff
@@ -1074,8 +1074,14 @@ void AddSarc(int i)
 	dObject[Objects[i].position.x][Objects[i].position.y - 1] = -(i + 1);
 	Objects[i]._oVar1 = GenerateRnd(10);
 	Objects[i]._oRndSeed = AdvanceRndSeed();
-	if (Objects[i]._oVar1 >= 8)
-		Objects[i]._oVar2 = PreSpawnSkeleton();
+	if (Objects[i]._oVar1 >= 8) {
+		Monster *monster = PreSpawnSkeleton();
+		if (monster != nullptr) {
+			Objects[i]._oVar2 = monster->getId();
+		} else {
+			Objects[i]._oVar1 = -1;

```

Comment:
```suggestion
			Objects[i]._oVar2 = -1;
```

Reply:
Or was the intent here to set `_oVar1` to a number less then 8 to avoid the later loading, with out checking that `_oVar2` is out of range? If that is the case then it should indeed be set to 0, but a comment should be added to make this clear.

Reply:
Yeah, that was the original intent. But I think that this approach is more clear, so let's go ahead with it 😄